### DISCUSSION TITLE
Break the inheritance

### DISCUSSION THREAD

**Discussion by GeorgeGeorgeGe** [rejetto/hfs#216] [Q&A]
Hello everybody,
Would you please tell me if we can break the sharing permissions inherited from parent folder added from disk so we can assign new permissions on the child folders, the same way we do in windows sharing permissions ?
let us say that i have the folders: "1" , "2" and "3" under the home directory. I want "user1" to be able to see the folders: "1" and "3" only and "user2" able to see folder "2" only. is that possible and how?
Please advice.

Thanks in advance.

**Comment by rejetto** [comment]
so, you have folder X from disk, and 1/2/3 inside.
I understand you are wondering how to configure these folders since they don't appear in the VFS (because they are dynamically loaded from folder X, on disk).

## Easy method
in the VFS, click folder X, then add folders 1/2/3 (from disk), so they now appear in the VFS, and you can freely edit permissions.

## Complex but powerful
Use the "masks" field.
This field is quite powerful but a bit complex. I'll try to provide an easier editor in the future.
You can find it briefly explained here https://github.com/rejetto/hfs/blob/main/config.md#virtual-file-system-vfs

The "trick" is that a mask doesn't necessarily have to contain wildcards, but you can just enter a simple name, like "1".

So my suggestion is, go to folder X and enter this in the masks field
```yaml
1|3:
  can_see: [user1]
2:
  can_see: [user2]
```

**Reply by GeorgeGeorgeGe** [reply]
Thanks for your quick reply, i really appreciate it.
what if i have the folder "3" inside folder "2" that is insider folder "1" on the disk and i want "user1" to be able to see folder "1" and "3" only, and "user 2" can see folder "2" only . what is the best way to do that please.

Thanks in advance

**Comment by rejetto** [comment]
easy method: you first add folder 1, then add 2 inside 1, and then 3 inside 2.
At this point you set whatever permission in each folder.

masks method:
```yaml
1:
  can_see: [user1]
1/2/3:
  can_see: [user1]
1/2:
  can_see: [user2]
```


I guess your example is not a real case 🙃

**Reply by GeorgeGeorgeGe** [reply]
trust me it is a real scenario that i am working on.
based on your reply above, "user1" will be able to see folder "2" from listing folder "1" itself because he has permissions on it that will be inherited to child folder under it, that means i am unable to prevent "user1" from seeing what is inside folder "2", correct ?
Thanks for you help

**Comment by rejetto** [comment]
nope, the masks method i described is complete and already does what you asked (or should).

In the easy method I wrote that you set whatever permission. Once you create your structure (1+2+3), you click on 2, and decide WHO can see folder 2, so you set user2 there. This way user1 won't see just folder 2

Permissions are inherited unless you are specifying differently.

**Reply by GeorgeGeorgeGe** [reply]
Yes but in this case "user1" can not move to folder "3" because he is unable to go there through folder "2" because he can not see it.
i am trying to display on the web the same structure that is on the disk.
folder are inside each others on the disk and on the VFS.
please advise.
Thanks

**Comment by rejetto** [comment]
Ok, but your problem is inherent to the description of the task, since you said that user1 must not see folder 2, and yet you want user1 to navigate without seeing?
I can try to guess: maybe you don't need to prevent from "seeing" but just from downloading?
And still, you can consider providing a direct link to 3, somewhere. Such link would bring user1 to folder 3.

**Reply by GeorgeGeorgeGe** [reply]
Yes, you are right, i should provide a direct link to folder "3"
Thank you very much for your help, i really appreciate it.

**Reply by rejetto** [reply]
you're welcome, and let me know in case something doesn't work as I said

**Reply by GeorgeGeorgeGe** [reply]
I will, Thanks again.
